app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt

```diff
@@ -174,7 +178,10 @@ fun SignInScreen(
                   Text("Sign in with email and password")
                 }
           }
-          if (currentUser != null) {
+          if (currentUser != null &&
+              !signInCompleted
+                  .value) { // extra condition on sign-in completion to avoid unnecessary
+            // recompositions
```

Comment:
I don't understand why this condition is necessary ? Can you elaborate please

Reply:
After private conversation, this line is necessary to avoid getting a new token when the log out is not completely finalised.